Documentation missing on "import.meta.hot" magic

### Describe the bug

Here's a pretty normal self-accepting HMR module:

```javascript
if (import.meta.hot) {
  import.meta.hot.accept(() => {
    if (cannotHandleUpdate) {
      import.meta.hot.invalidate();
    }
  });
}
```

Unfortunately, TypeScript doesn't like this:

```
src/main.ts:5:7 - error TS2532: Object is possibly 'undefined'.
```

This is a correct error message, because the `hot` property could be unset between the check and the time the callback is invoked (theoretically). A simple fix is to store the known-good value:

```typescript
if (import.meta.hot) {
  const hot = import.meta.hot;
  hot.accept(() => {
    if (cannotHandleUpdate) {
      hot.invalidate();
    }
  });
}
```

This fixes the TypeScript error, but **has entirely different behavior**, due to Vite ([related issue](https://github.com/vitejs/vite/issues/9235#issuecomment-1189878048)).

# Requested change

The documentation, specifically [this section](https://vitejs.dev/guide/api-hmr.html#hot-accept-cb) should add details that the text `import.meta.hot.accept(` must appear in the source code for the module to be self-accepting.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-bkzjrj?file=src%2Fmain.ts,index.html&terminal=dev

### Steps to reproduce

_No response_

### System Info

```shell
`vite@4.0.4`
```


### Used Package Manager

npm

### Logs

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/docs/guide/api-hmr.md
+++ b/docs/guide/api-hmr.md
@@ -70,9 +70,9 @@ if (import.meta.hot) {
 
 A module that "accepts" hot updates is considered an **HMR boundary**.
 
-Note that Vite's HMR does not actually swap the originally imported module: if an HMR boundary module re-exports imports from a dep, then it is responsible for updating those re-exports (and these exports must be using `let`). In addition, importers up the chain from the boundary module will not be notified of the change.
+Vite's HMR does not actually swap the originally imported module: if an HMR boundary module re-exports imports from a dep, then it is responsible for updating those re-exports (and these exports must be using `let`). In addition, importers up the chain from the boundary module will not be notified of the change. This simplified HMR implementation is sufficient for most dev use cases, while allowing us to skip the expensive work of generating proxy modules.
 
-This simplified HMR implementation is sufficient for most dev use cases, while allowing us to skip the expensive work of generating proxy modules.
+Vite requires that the call to this function appears as `import.meta.hot.accept(` (whitespace-sensitive) in the source code in order for the module to accept update. This is a requirement of the static analysis that Vite does to enable HMR support for a module.
 
 ## `hot.accept(deps, cb)`
 

